### PR TITLE
Refresh for v1.8.4

### PR DESCRIPTION
**Pull request by tdoublep**
- Update package version
- Update release notes

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -3,6 +3,25 @@ Release Notes
...
+- Fix bug related to number of features-based optimization in compressed ensemble.
+
+New features:
+
+- ExtraTrees support in inference engine.
+- New features for knowledge distillation.
+
+Perf. improvements:
+
+- Performance improvement for decision trees.
```
Comment (andreea-anghel) [inline docs/source/release_notes.rst:23]: are these inference-related improvements? do they refer to ensemble models or just individual tree models?

Comment (tdoublep) [reply docs/source/release_notes.rst:23]: They are training-related I believe and affect all tree-based models

Comment (tdoublep) [reply docs/source/release_notes.rst]: Updated release notes to reflect this.

**Review by andreea-anghel — APPROVED** [review]
LGTM

**Pull request merged**